client/post-editor/post-editor.jsx

```diff
@@ -820,6 +816,11 @@ export const PostEditor = React.createClass( {
 
 	onSaveSuccess: function( message, action, link ) {
 		const post = PostEditStore.get();
+		const isNotPrivateOrIsConfirmed = ( 'private' !== post.status ) || ( 'closed' !== this.state.confirmationSidebar );
```

Comment:
Thanks. It looks like we probably shouldn't be entering "publishing" state for the confirmation sidebar if we weren't in "open" state to begin with -- so I fixed this by preventing that state from being reached.